### PR TITLE
To latex position

### LINKED ISSUE #35281: ENH: to_latex positional argument
#### Is your feature request related to a problem?

I would like to have the possibility to specify the location of the table I'm saving, or instead of having 
```
\begin{table}
...
\end{table}
```

be able to have 
```
\begin{table}[h]
...
\end{table}
```

#### Describe the solution you'd like

I guess the solution would be to add another argument in the `to_latex` API which could take values in `(None, 'h', 'H', 'h!'...)`

#### Describe alternatives you've considered

I'm saving my tables to `.tex` files and then use the `\input{path/to/tex}` command, I did not find the possibility to give the [h] argument with input...

### PR DESCRIPTION
**Pull request by SylvainLan**
- [ ] closes #35281 
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/io/formats/test_to_latex.py
Comment (WillAyd) [inline pandas/tests/io/formats/test_to_latex.py:486]: Can you create this as its own test? Same comment with one below

**Review by WillAyd — COMMENTED** [review]
Thanks for the PR. Minor comment on test but otherwise I think this is OK

**Comment by SylvainLan** [comment]
Hello @WillAyd , thanks for the review ! 

I did the changes you asked for, but one test seems to have failed, I'm not sure it's related to what I've written

Would you also want tests when the caption, the label and the position is given ?

**Comment by WillAyd** [comment]
Just restarted the failure let's see

**Comment by WillAyd** [comment]
@toobaz

File: pandas/io/formats/latex.py
Comment (toobaz) [inline pandas/io/formats/latex.py:295]: Any obvious reason why this was moved up here? I would find `` buf.write(f"\\begin{{table}}{position_}")`` more readable that the ``write`` below.

File: pandas/io/formats/latex.py
Comment (toobaz) [inline pandas/io/formats/latex.py:293]: As this test grows, it would be better to have ``self._table_float = any([p is not None for p in (caption, label, position)])`` inside ``__init__.py``, to be used both here and in ``_write_tabular_end``.

**Review by toobaz — COMMENTED** [review]
Mostly looks good, two small comments (but I wouldn't object to solving the ``self._table_float`` one in a separate PR).

File: pandas/io/formats/latex.py
Comment (SylvainLan) [reply pandas/io/formats/latex.py:295]: The reason I did this was only because the line was getting too long, but I can write it as it was for sure. 
Maybe we should unify the way it's done in `_write_tabular_begin` and `_write_longtable_begin` ? But it may be nitpicking

Comment (toobaz) [reply pandas/io/formats/latex.py]: > The reason I did this was only because the line was getting too long, but I can write it as it was for sure.

I do find it more readable - but if you want to break the line, you could do so at ``\n``, so after the position.

> Maybe we should unify the way it's done in `_write_tabular_begin` and `_write_longtable_begin` ?

Wow... indeed the difference between `_write_tabular_begin` and `_write_longtable_begin` seems to amount to the environment name: unless I'm mistaken, the merge is definitely worth doing (same for ``_end``). By the way, if the function are unified feel free to move back the ``any([p is not None...`` down there.

**Comment by SylvainLan** [comment]
@toobaz I tried to match the two functions `_write_table_begin` and `_write_longtable_begin`, but I don't think there is a need for the `_table_float` here, because:
- There is no change in the begin line: whether or not a caption, label or position is given, it will always be `\begin{longtable}...`
- The only difference would be for the line `buf.write("\\\\\n")` but here again the `_table_float` argument is not relevant because this behaviour is needed when a caption or a label is provided, but if only the position would be given it shouldn't be needed

File: pandas/io/formats/latex.py
Comment (toobaz) [inline pandas/io/formats/latex.py:365]: Duplicated lines?

**Comment by toobaz** [comment]
> I tried to match the two functions

My suggestion was actually to entirely suppress ``_write_longtable_begin``, given that the only implications of ``longtable=True`` are that the float must be produced, and that its environment name must be ``longtable`` rather than ``table``: all the rest is the same as ``_write_tabular_begin`` (same for ``*_end``).

But this is not a strict necessity, at least in this PR. So if you prefer we can merge this PR after solving my last comment above.

File: pandas/io/formats/latex.py
Comment (SylvainLan) [reply pandas/io/formats/latex.py]: Right thanks !

**Comment by SylvainLan** [comment]
> My suggestion was actually to entirely suppress `_write_longtable_begin`

OK @toobaz I hadn't understood ! I tried doing something to resolve this, but I suggest doing it on a separate PR because I think it will get a bit messy.

This work is currently on a [branch of mine](https://github.com/SylvainLan/pandas/blob/merge_table_longtable/pandas/io/formats/latex.py) and we can look at it once this PR is merged

**Comment by toobaz** [comment]
Looks ready to merge to me. @WillAyd any idea on the two checks still failing?

**Comment by simonjayhawkins** [comment]
> Looks ready to merge to me. @WillAyd any idea on the two checks still failing?

see #35481. there are 20 fails here, should be only 4 now with lastest numpy-dev

**Review by WillAyd — APPROVED** [review]
lgtm @jreback

**Comment by WillAyd** [comment]
Thanks @SylvainLan

**Pull request merged**
